src/Native/Unix/CMakeLists.txt

```diff
@@ -231,3 +231,7 @@ add_subdirectory(System.Native)
 add_subdirectory(System.Net.Http.Native)
 add_subdirectory(System.Net.Security.Native)
 add_subdirectory(System.Security.Cryptography.Native)
+
+if(OSX)
+add_subdirectory(System.Security.Cryptography.Native.Apple)
```

Comment:
In this case it's about the provider instead of the OS; though they're similar.  OpenSSL runs on OSX (though, as we've seen, distribution is not easy), so calling it "OSX" doesn't indicate that we're using Apple's cryptography libraries instead of OpenSSL.

The other complication is that Apple's crypto stuff is spread across a couple of different  names, and so "Apple" was the best unifier I had to be equivalent to "OpenSsl".